### ISSUE TITLE
auth.secret_name provided but still fails with "authentication required" on Docker Hub private repo / cosign

### ISSUE THREAD

**Issue by xopham** [sse-secure-systems/connaisseur#575]
### Discussed in https://github.com/sse-secure-systems/connaisseur/discussions/573

<div type='discussions-op-text'>

<sup>Originally posted by **funkypenguin** March  7, 2022</sup>
Hey folks!

I'm trying to use a cosign validator against my private Docker Hub repos, but I've not been able to get connaisseur to authenticate - so I get errors like this when testing with `kubectl run`:

```
Error from server: admission webhook "connaisseur-svc.connaisseur.svc" denied the request: Unexpected Cosign exception for image "docker.io/myord/curl:latest": Error: GET https://index.docker.io/v2/myorg/curl/manifests/latest: UNAUTHORIZED: authentication required; [map[Action:pull Class: Name:gergorg/curl Type:repository]]
main.go:46: error during command execution: GET https://index.docker.io/v2/myorg/curl/manifests/latest: UNAUTHORIZED: authentication required; [map[Action:pull Class: Name:myorg/curl Type:repository]]
```

My validator is defined like this:

```
  - name: default
    type: cosign  # or other supported validator (e.g. "cosign")
    # host: notary.docker.io # in case of notaryv1, configure the notary server to be used
    auth: 
      secret_name: registry-credentials  # reference a k8s secret in the form required by the validator type (check the docs)
    trust_roots:
    # # the `default` key is used if no key is specified in image policy
    - name: default
      key: |  # enter your public key below (my note : the one below is a dummy one used for dev)
        -----BEGIN PUBLIC KEY-----
        MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEizCL5gnS98PS929yPNo61x7MSuhs
        g4qfWtLxfaRQfy2YPhZ2//y+AeaGLU9Y98jQHyZqLON4F+9DFJqX2RbwFQ==
        -----END PUBLIC KEY-----
    #  # or (only for notaryv1 validator)
    #  username: myuser
    #  password: mypass
```

And my I've confirmed that the credentials are correctly mounted into the pod container at `/app/connaisseur-config/default/.docker/config.json`.

When I set `LOGLEVEL=DEBUG`, I see the following output from connaisseur:

```
[2022-03-07 01:52:23,831] INFO: {'message': 'Unexpected Cosign exception for image "docker.io/myorg/curl:latest": Error: GET https://index.docker.io/v2/myorg/curl/manifests/latest: UNAUTHORIZED: authentication required; [map[Action:pull Class: Name:myorg/curl Type:repository]]\nmain.go:46: error during command execution: GET https://index.docker.io/v2/myorg/curl/manifests/latest: UNAUTHORIZED: authentication required; [map[Action:pull Class: Name:myorg/curl Type:repository]]\n.', 'context': {'trust_data_type': 'dev.cosignproject.cosign/signature', 'stderr': 'Error: GET https://index.docker.io/v2/myorg/curl/manifests/latest: UNAUTHORIZED: authentication required; [map[Action:pull Class: Name:myorg/curl Type:repository]]\nmain.go:46: error during command execution: GET https://index.docker.io/v2/myorg/curl/manifests/latest: UNAUTHORIZED: authentication required; [map[Action:pull Class: Name:myorg/curl Type:repository]]\n', 'image': 'docker.io/myorg/curl:latest', 'trust_root': 'default', 'detection_mode': False}}
[2022-03-07 01:52:23,832] DEBUG: No alerting configuration file found.
[2022-03-07 01:52:23,832] ERROR: {'message': 'Unexpected Cosign exception for image "docker.io/myorg/curl:latest": Error: GET https://index.docker.io/v2/myorg/curl/manifests/latest: UNAUTHORIZED: authentication required; [map[Action:pull Class: Name:myorg/curl Type:repository]]\nmain.go:46: error during command execution: GET https://index.docker.io/v2/myorg/curl/manifests/latest: UNAUTHORIZED: authentication required; [map[Action:pull Class: Name:myorg/curl Type:repository]]\n.', 'context': {'trust_data_type': 'dev.cosignproject.cosign/signature', 'stderr': 'Error: GET https://index.docker.io/v2/myorg/curl/manifests/latest: UNAUTHORIZED: authentication required; [map[Action:pull Class: Name:myorg/curl Type:repository]]\nmain.go:46: error during command execution: GET https://index.docker.io/v2/myorg/curl/manifests/latest: UNAUTHORIZED: authentication required; [map[Action:pull Class: Name:myorg/curl Type:repository]]\n', 'image': 'docker.io/myorg/curl:latest', 'trust_root': 'default', 'detection_mode': False, 'user': 'system:admin', 'operation': 'CREATE', 'kind': 'Pod', 'name': 'test', 'namespace': 'sandbox'}}
```

Any ideas? :)

Thanks,
D
</div>

**Comment by xopham** [comment]
In `v2.5.0`, the handling of environmental variables was changed which introduced a bug by which the env parameter is unset and consequently setting of DOCKER_CONFIG fails: https://github.com/sse-secure-systems/connaisseur/blob/master/connaisseur/validators/cosign/cosign_validator.py#L232

**Comment by xopham** [comment]
fixed by #576 and released in `v2.5.1`
